### PR TITLE
Parameters support on RunJS

### LINKED ISSUE #4274: js queries parameters
### Is your proposal related to a problem?

If you will add this feature, everyone will be able to use queries like a js functions - devide code on smaller parts and reuse some parts multiple times

Right now I can do that, but this solution SO UGLY. I can use some global variable in some js query, and to change this variable before I start the query

### Describe the solution you'd like

I want to have ability to run any query with additional parameters, like in Retool. 
Rettol example:

somequery.trigger({additionalScope: {customprop1: 100}});

### PR DESCRIPTION
**Pull request by johnsoncherian**
Fixes #4274

### CODE REVIEW

File: frontend/src/Editor/Inspector/ActionConfigurationPanels/RunjsParamters.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { CodeHinter } from '../../CodeBuilder/CodeHinter';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Inspector/ActionConfigurationPanels/RunjsParamters.jsx:2]:
```suggestion
import { CodeHinter } from '@/Editor/CodeBuilder/CodeHinter';
```

File: frontend/src/Editor/QueryManager/QueryEditors/Runjs/ParameterForm.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  }, [showModal]);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    onSubmit && onSubmit({ name, defaultValue });
+  };
+
+  useEffect(() => {
+    console.log(otherParams, name);
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/QueryManager/QueryEditors/Runjs/ParameterForm.jsx:38]:
```suggestion
```

File: frontend/src/_helpers/appUtils.js [hunk 3]
```diff
@@ -399,9 +399,16 @@ function executeActionWithDebounce(_ref, event, mode, customVariables) {
 
       case 'run-query': {
         const { queryId, queryName } = event;
+        const params = event['parameters'];
+        const resolvedParams = {};
+        if (params) {
+          Object.keys(params).map(
+            (k) => (resolvedParams[k] = resolveReferences(params[k], _ref.state.currentState, undefined))
```
Comment (kavinvenkatachalam) [inline frontend/src/_helpers/appUtils.js:406]: Instead of `k`, can we have something like `param`

File: frontend/src/_helpers/appUtils.js [hunk 4]
```diff
@@ -821,7 +828,7 @@ export function getQueryVariables(options, state) {
   return queryVariables;
 }
 
-export function previewQuery(_ref, query, calledFromQuery = false) {
+export function previewQuery(_ref, query, calledFromQuery = false, parameters, hasParamSupport) {
```
Comment (kavinvenkatachalam) [inline frontend/src/_helpers/appUtils.js:831]: Can we add some default value to the newly added params?

File: frontend/src/_helpers/appUtils.js [hunk 5]
```diff
@@ -911,7 +936,7 @@ export function previewQuery(_ref, query, calledFromQuery = false) {
   });
 }
 
-export function runQuery(_ref, queryId, queryName, confirmed = undefined, mode = 'edit') {
+export function runQuery(_ref, queryId, queryName, confirmed = undefined, mode = 'edit', parameters) {
```
Comment (kavinvenkatachalam) [inline frontend/src/_helpers/appUtils.js:939]: Can we add some default value to the newly added params?

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
Hey @johnsoncherian - Found few issues,

1) Able to create params with the same name
2) Getting the unsaved popup without making any changes on runJS

**Review by kavinvenkatachalam — APPROVED** [review]

**Review by arpitnath — CHANGES_REQUESTED** [review]

File: frontend/src/Editor/QueryManager/QueryEditors/Runjs/ParameterForm.jsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    }
+  }, [showModal]);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    onSubmit && onSubmit({ name, defaultValue });
+  };
+
+  useEffect(() => {
+    if (!isValidVariableName(name)) {
```
[the remark this answers is not in the document]
Comment (johnsoncherian) [reply frontend/src/Editor/QueryManager/QueryEditors/Runjs/ParameterForm.jsx:38]: Added check added in page submission level.

**Review by arpitnath — APPROVED** [review]

**Pull request merged**